store/storage/pebbledb/db_test.go

```diff
@@ -0,0 +1,364 @@
+package pebbledb
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	storeKey1 = "store1"
+)
+
+func TestDatabase_Close(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	require.NoError(t, db.Close())
+}
+
+func TestDatabase_LatestVersion(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	lv, err := db.GetLatestVersion()
+	require.Error(t, err)
+	require.Zero(t, lv)
+
+	for i := uint64(1); i <= 1001; i++ {
+		err = db.SetLatestVersion(i)
+		require.NoError(t, err)
+
+		lv, err = db.GetLatestVersion()
+		require.NoError(t, err)
+		require.Equal(t, i, lv)
+	}
+}
+
+func TestDatabase_CRUD(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	ok, err := db.Has(storeKey1, 1, []byte("key"))
+	require.NoError(t, err)
+	require.False(t, ok)
+
+	err = db.Set(storeKey1, 1, []byte("key"), []byte("value"))
+	require.NoError(t, err)
+
+	ok, err = db.Has(storeKey1, 1, []byte("key"))
+	require.NoError(t, err)
+	require.True(t, ok)
+
+	val, err := db.Get(storeKey1, 1, []byte("key"))
+	require.NoError(t, err)
+	require.Equal(t, []byte("value"), val)
+
+	err = db.Delete(storeKey1, 1, []byte("key"))
+	require.NoError(t, err)
+
+	ok, err = db.Has(storeKey1, 1, []byte("key"))
+	require.NoError(t, err)
+	require.False(t, ok)
+
+	val, err = db.Get(storeKey1, 1, []byte("key"))
+	require.NoError(t, err)
+	require.Nil(t, val)
+}
+
+func TestDatabase_VersionedKeys(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	for i := uint64(1); i <= 100; i++ {
+		err := db.Set(storeKey1, i, []byte("key"), []byte(fmt.Sprintf("value%03d", i)))
+		require.NoError(t, err)
+	}
+
+	for i := uint64(1); i <= 100; i++ {
+		bz, err := db.Get(storeKey1, i, []byte("key"))
+		require.NoError(t, err)
+		require.Equal(t, fmt.Sprintf("value%03d", i), string(bz))
+	}
+}
+
+func TestDatabase_GetVersionedKey(t *testing.T) {
+	db, err := New(t.TempDir())
+	require.NoError(t, err)
+	defer db.Close()
+
+	// store a key at version 1
+	err = db.Set(storeKey1, 1, []byte("key"), []byte("value"))
+	require.NoError(t, err)
+
+	// assume chain progresses to version 10 w/o any changes to key
```

Comment:
can we add a case that it's been set at version 1 and 11, then query for version 10, make sure it gets the value at version 1.

Reply:
Done :)